packages/react-server-dom-webpack/package.json

```
@@ -50,7 +50,7 @@
   "peerDependencies": {
     "react": "^17.0.0",
     "react-dom": "^17.0.0",
-    "webpack": "^4.43.0"
+    "webpack": "^5"
```

I think we need to downgrade this back to Webpack 4, e.g.
```suggestion
    "webpack-4": "^4"
```
@sebmarkbage says this plug-in doesn’t work with Webpack 5 since it uses internals and plug-in apis and needs to be ported. (And there's currently no automated test coverage to verify this does or doesn't work.)

**Edit 1** Apparently we can test this by building and then `yarn start`ing the fixture in `/fixtures/flight/`.

The built bundle has a peer dependency on Webpack 4 (via the workspaces) and this might cause problems. I'll test this.

**Edit 2**: @sebmarkbage has also expressed concern that this change would break this demo: https://github.com/reactjs/server-components-demo

---

Can we rollback the peer dependency change for this package only?

---

I've tested the flight fixture after this most recent change by...
```sh
# root
yarn install
yarn build

# fixtures/flight
yarn install
yarn start
```

And then loading [localhost:3000](http://localhost:3000/) which shows the following warnings/errors:
```
Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead. Until you switch to the new API, your app will behave as if it's running React 17. Learn more: https://reactjs.org/link/switch-to-createroot

Failed to load resource: net::ERR_CONNECTION_REFUSED

Uncaught TypeError: Failed to fetch
  The above error occurred in the <Content> component:

    at Content
    at Suspense

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
```

The first warning is expected and indicates that the fixture is just a little out of date. The following errors seem bad though.

For comparison, I've tested the same steps on HEAD of main branch, and I see the app load successfully. So it seems like this update certainly breaks the _flight fixture_ if not the downstream packages/dependencies.

cc @sebmarkbage to confirm is this is _acceptable_ for the time being?

---

The fixture is acceptable but we need to fix the published package. Worst case maybe we can turn this to `private: true` and exclude it from future builds to avoid confusion until it's fixed.

---

Just to be clear, we're commenting on an outdated version of this PR here. The latest one makes no changes to the `packages/react-server-dom-webpack/package.json` package (or its peer dependencies)

---

Sorry for the drive-by. 

> says this plug-in doesn’t work with Webpack 5 since it uses internals and plug-in apis and needs to be ported. (And there's currently no automated test coverage to verify this does or doesn't work.)

I wrote a PR a while ago to support both webpack 5 and webpack 4 (https://github.com/facebook/react/pull/20524/files) in the `react-dom-server-webpack` plugin package. 

If there is interest, I would be happy to pick it back up and get it into a state where it could be merged in? It sounds like we'd want to add automated tests for the plugin before merging the linked PR.